test/sequential/proffOfStake.3rdPartyRewards.burning.test.ts

```diff
@@ -0,0 +1,215 @@
+          Assets.DEFAULT_AMOUNT.divn(2),
+          Assets.DEFAULT_AMOUNT,
+        ),
+      ),
+      Sudo.sudoAs(
+        testUser3,
+        Xyk.mintLiquidity(
+          MGA_ASSET_ID,
+          newToken,
+          Assets.DEFAULT_AMOUNT.divn(2),
+          Assets.DEFAULT_AMOUNT,
+        ),
+      ),
+      Sudo.sudo(
+        await Vesting.forceVested(
+          sudo.keyRingPair.address,
+          testUser4,
+          Assets.DEFAULT_AMOUNT.divn(2),
+          MGA_ASSET_ID,
+          100,
+        ),
+      ),
+      Assets.promotePool(liqId.toNumber(), 20),
+      Sudo.sudoAs(
+        testUser4,
+        Xyk.mintLiquidityUsingVested(
+          newToken,
+          Assets.DEFAULT_AMOUNT.divn(2),
+          Assets.DEFAULT_AMOUNT,
+        ),
+      ),
+      Sudo.sudoAs(
+        testUser4,
+        MPL.reserveVestingLiquidityTokensByVestingIndex(liqId),
+      ),
+    );
+    testLog.getLog().info("liqId: " + liqId.toString());
+  });
+
+  describe("Burn liquidity", () => {
+    beforeAll(async () => {
+      await Sudo.batchAsSudoFinalized(
+        ...(await rewardAndActivatePool(
+          newToken,
+          testUser0,
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          null,
+        )),
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+    });
+    test("GIVEN a user with N tokens activated for 3rd party, WHEN the user wants to burn ( all the existing - N ) THEN the operation works", async () => {
+      const testUser = testUser1;
+      const amountToActivate = Assets.DEFAULT_AMOUNT.divn(10);
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          amountToActivate,
+          newToken,
+          null,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const amountToBurn = (await getBalanceOfAsset(liqId, testUser)).free.sub(
+        amountToActivate,
+      );
+      await signTx(
+        getApi(),
+        Xyk.burnLiquidity(newToken, MGA_ASSET_ID, amountToBurn),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+      expect(amountToBurn).bnGt(BN_ZERO);
+      expect((await getBalanceOfAsset(liqId, testUser)).free).bnGt(BN_ZERO);
+      expect((await getBalanceOfAsset(liqId, testUser)).reserved).bnEqual(
+        amountToActivate,
+      );
+      expect(
+        await ProofOfStake.activatedLiquidityForSchedules(
+          liqId,
+          testUser.keyRingPair.address,
+          newToken,
+        ),
+      ).bnEqual(amountToActivate);
+    });
+  });
+});
+async function rewardAndActivatePool(
```

Comment:
We use this function twice. Should we add it to some shared file?

Reply:
done! moved to ProofOfStake class